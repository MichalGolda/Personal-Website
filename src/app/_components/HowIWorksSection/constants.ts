export enum PhaseCardColors {
  PHASE_FIRST = "before:bg-[#30E6FF] text-[#30E6FF]",
  PHASE_SECOND = "before:bg-[#FFB930] text-[#FFB930]",
  PHASE_THIRD = "before:bg-[#FF4930] text-[#FF4930]",
  PHASE_FOURTH = "before:bg-[#B130FF] text-[#B130FF]",
}
