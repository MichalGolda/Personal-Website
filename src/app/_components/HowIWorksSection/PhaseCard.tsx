import { PhaseCardColors } from "./constants";

export type PhaseCardProps = {
  title: string;
  headline: string;
  body: string;
  color: PhaseCardColors;
  className?: string;
};

const PhaseCard: React.FC<PhaseCardProps> = ({
  title,
  headline,
  color,
  body,
  className,
  ...props
}) => {
  return (
    <div
      className={`flex flex-col gap-y-4 border border-solid border-lightGrey p-4 bg-white ${className}`}
      {...props}
    >
      <div className="flex flex-col">
        <span
          className={`before:block flex items-center gap-x-2 text-sm md:text-base font-semibold uppercase before:w-2 before:h-2 ${color}`}
        >
          {headline}
        </span>
        <h4 className="text-lg md:text-xl lg:text-2xl text-secondary font-bold text-secondary">
          {title}
        </h4>
      </div>
      <p className="text-body text-sm md:text-base">{body}</p>
    </div>
  );
};

export default PhaseCard;
