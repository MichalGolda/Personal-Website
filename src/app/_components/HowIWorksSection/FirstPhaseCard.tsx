import PhaseCard from "./PhaseCard";
import { PhaseCardColors } from "./constants";

const FirstPhaseCard = () => (
  <PhaseCard
    className="w-full md:w-1/2"
    headline="faza 1"
    title="Określenie oczekiwań klienta"
    color={PhaseCardColors.PHASE_FIRST}
    body="W pierwszej fazie dokładnie analizuję oczekiwania klienta, te informacje stanowią fundament, na którym opiera się proces przygotowania ostatecznego rozwiązania."
    data-aos="fade-right"
    data-aos-duration="500"
  />
);

export default FirstPhaseCard;
