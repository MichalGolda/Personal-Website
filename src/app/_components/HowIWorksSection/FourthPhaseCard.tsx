import PhaseCard from "./PhaseCard";
import { PhaseCardColors } from "./constants";

const FourthPhaseCard = () => (
  <PhaseCard
    className="w-full md:w-1/2 md:ml-[calc((50%/3)*3)]"
    headline="faza 4"
    title="Wdrożenie"
    color={PhaseCardColors.PHASE_FOURTH}
    body="Po zatwierdzeniu projektu przechodzę do wdrożenia na serwer produkcyjny. Dodatkowo integruję narzędzia takie jak Google Analytics, Sentry, Google Search Console. Przeprowadzam również kompleksową optymalizację SEO."
    data-aos="fade-right"
    data-aos-duration="2000"
  />
);

export default FourthPhaseCard;
