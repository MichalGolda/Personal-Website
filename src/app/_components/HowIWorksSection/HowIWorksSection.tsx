"use client";

import { useInViewSection } from "@/app/_context/inViewSectionContext";
import { Container, SectionHeading } from "@/app/_components";
import { PhaseCard } from "./PhaseCard";

export default function HowIWorksSection() {
  const { ref } = useInViewSection("#how-i-works");

  return (
    <div ref={ref} className="my-64">
      <Container>
        <SectionHeading headline="proces" title="Jak działam?" />
        <div className="flex flex-col gap-y-16 w-full mt-32">
          <PhaseCard
            className="w-full md:w-1/2"
            headline="faza 1"
            title="Określenie oczekiwań klienta"
            color="PHASE_ONE"
            body="W pierwszej fazie dokładnie analizuję oczekiwania klienta, te nformacje stanowią fundament, na którym opiera się proces przygotowania ostatecznego rozwiązania."
            data-aos="fade-right"
            data-aos-duration="500"
          />
          <PhaseCard
            className="w-full md:w-1/2 md:ml-[calc(50%/3)]"
            headline="faza 2"
            title="Design UX/UI"
            color="PHASE_TWO"
            body="Na bazie zebranych informacji tworzę projekt graficzny. W trakcie projektowania regularnie konsultuję się z klientem, aby na bieżąco uwzględniać wprowadzane zmiany. Przechodzę do kolejnego etapu dopiero po uzyskaniu pełnej akceptacji."
            data-aos="fade-right"
            data-aos-duration="1000"
          />
          <PhaseCard
            className="w-full md:w-1/2 md:ml-[calc((50%/3)*2)]"
            headline="faza 3"
            title="Budowa"
            color="PHASE_THREE"
            body="W trzeciej fazie przystępuję do budowy rozwiązania, począwszy od kodowania aż po testowanie. Zapewniam, że każdy element jest starannie sprawdzany i testowany, aby dostarczyć wysokiej jakości produkt."
            data-aos="fade-right"
            data-aos-duration="1500"
          />
          <PhaseCard
            className="w-full md:w-1/2 md:ml-[calc((50%/3)*3)]"
            headline="faza 4"
            title="Wdrożenie"
            color="PHASE_FOUR"
            body="Po zatwierdzeniu projektu przechodzę do wdrożenia na serwer produkcyjny. Dodatkowo integruję narzędzia takie jak Google Analytics, Sentry, Google Search Console. Przeprowadzam również kompleksową optymalizację SEO."
            data-aos="fade-right"
            data-aos-duration="2000"
          />
        </div>
      </Container>
    </div>
  );
}
