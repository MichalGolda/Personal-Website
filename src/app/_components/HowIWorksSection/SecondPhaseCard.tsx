import PhaseCard from "./PhaseCard";
import { PhaseCardColors } from "./constants";

const SecondPhaseCard = () => (
  <PhaseCard
    className="w-full md:w-1/2 md:ml-[calc(50%/3)]"
    headline="faza 2"
    title="Design UX/UI"
    color={PhaseCardColors.PHASE_SECOND}
    body="Na bazie zebranych informacji tworzę projekt graficzny. W trakcie projektowania regularnie konsultuję się z klientem, aby na bieżąco uwzględniać wprowadzane zmiany. Przechodzę do kolejnego etapu dopiero po uzyskaniu pełnej akceptacji."
    data-aos="fade-right"
    data-aos-duration="1000"
  />
);

export default SecondPhaseCard;
