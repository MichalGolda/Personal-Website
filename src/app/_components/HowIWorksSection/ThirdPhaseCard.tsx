import PhaseCard from "./PhaseCard";
import { PhaseCardColors } from "./constants";

const ThirdPhaseCard = () => (
  <PhaseCard
    className="w-full md:w-1/2 md:ml-[calc((50%/3)*2)]"
    headline="faza 3"
    title="Budowa"
    color={PhaseCardColors.PHASE_THIRD}
    body="W trzeciej fazie przystępuję do budowy rozwiązania, począwszy od kodowania aż po testowanie. Zapewniam, że każdy element jest starannie sprawdzany i testowany."
    data-aos="fade-right"
    data-aos-duration="1500"
  />
);

export default ThirdPhaseCard;
