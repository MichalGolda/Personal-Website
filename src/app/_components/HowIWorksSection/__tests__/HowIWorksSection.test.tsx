import "@/app/_mocks/intersectionObserver.mock";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import HowIWorksSection from "../HowIWorksSection";

describe("HowIWorksSection component", () => {
  it("renders unchanged", () => {
    const { container } = render(<HowIWorksSection />);

    expect(container).toMatchSnapshot();
  });
});
