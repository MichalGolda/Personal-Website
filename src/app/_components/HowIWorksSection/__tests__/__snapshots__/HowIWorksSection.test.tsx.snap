// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowIWorksSection component renders unchanged 1`] = `
<div>
  <div
    class="my-64"
  >
    <div
      class="md:max-w-[1024px] lg:max-w-[1320px] w-full mx-auto md:px-16 px-4 "
    >
      <div
        class="text-center max-w-3xl mx-auto"
      >
        <span
          class="text-primary font-semibold text-md md:text-lg lg:text-xl uppercase"
        >
          proces
        </span>
        <h2
          class="text-2xl md:text-3xl lg:text-4xl text-secondary font-bold"
        >
          Jak działam?
        </h2>
      </div>
      <div
        class="flex flex-col gap-y-16 w-full mt-32"
      >
        <div
          class="flex flex-col gap-y-4 border border-solid border-lightGrey p-4 bg-white w-full md:w-1/2"
          data-aos="fade-right"
          data-aos-duration="500"
        >
          <div
            class="flex flex-col"
          >
            <span
              class="before:block flex items-center gap-x-2 text-sm md:text-base font-semibold uppercase before:w-2 before:h-2 before:bg-[#30E6FF] text-[#30E6FF]"
            >
              faza 1
            </span>
            <h4
              class="text-lg md:text-xl lg:text-2xl text-secondary font-bold text-secondary"
            >
              Określenie oczekiwań klienta
            </h4>
          </div>
          <p
            class="text-body text-sm md:text-base"
          >
            W pierwszej fazie dokładnie analizuję oczekiwania klienta, te informacje stanowią fundament, na którym opiera się proces przygotowania ostatecznego rozwiązania.
          </p>
        </div>
        <div
          class="flex flex-col gap-y-4 border border-solid border-lightGrey p-4 bg-white w-full md:w-1/2 md:ml-[calc(50%/3)]"
          data-aos="fade-right"
          data-aos-duration="1000"
        >
          <div
            class="flex flex-col"
          >
            <span
              class="before:block flex items-center gap-x-2 text-sm md:text-base font-semibold uppercase before:w-2 before:h-2 before:bg-[#FFB930] text-[#FFB930]"
            >
              faza 2
            </span>
            <h4
              class="text-lg md:text-xl lg:text-2xl text-secondary font-bold text-secondary"
            >
              Design UX/UI
            </h4>
          </div>
          <p
            class="text-body text-sm md:text-base"
          >
            Na bazie zebranych informacji tworzę projekt graficzny. W trakcie projektowania regularnie konsultuję się z klientem, aby na bieżąco uwzględniać wprowadzane zmiany. Przechodzę do kolejnego etapu dopiero po uzyskaniu pełnej akceptacji.
          </p>
        </div>
        <div
          class="flex flex-col gap-y-4 border border-solid border-lightGrey p-4 bg-white w-full md:w-1/2 md:ml-[calc((50%/3)*2)]"
          data-aos="fade-right"
          data-aos-duration="1500"
        >
          <div
            class="flex flex-col"
          >
            <span
              class="before:block flex items-center gap-x-2 text-sm md:text-base font-semibold uppercase before:w-2 before:h-2 before:bg-[#FF4930] text-[#FF4930]"
            >
              faza 3
            </span>
            <h4
              class="text-lg md:text-xl lg:text-2xl text-secondary font-bold text-secondary"
            >
              Budowa
            </h4>
          </div>
          <p
            class="text-body text-sm md:text-base"
          >
            W trzeciej fazie przystępuję do budowy rozwiązania, począwszy od kodowania aż po testowanie. Zapewniam, że każdy element jest starannie sprawdzany i testowany.
          </p>
        </div>
        <div
          class="flex flex-col gap-y-4 border border-solid border-lightGrey p-4 bg-white w-full md:w-1/2 md:ml-[calc((50%/3)*3)]"
          data-aos="fade-right"
          data-aos-duration="2000"
        >
          <div
            class="flex flex-col"
          >
            <span
              class="before:block flex items-center gap-x-2 text-sm md:text-base font-semibold uppercase before:w-2 before:h-2 before:bg-[#B130FF] text-[#B130FF]"
            >
              faza 4
            </span>
            <h4
              class="text-lg md:text-xl lg:text-2xl text-secondary font-bold text-secondary"
            >
              Wdrożenie
            </h4>
          </div>
          <p
            class="text-body text-sm md:text-base"
          >
            Po zatwierdzeniu projektu przechodzę do wdrożenia na serwer produkcyjny. Dodatkowo integruję narzędzia takie jak Google Analytics, Sentry, Google Search Console. Przeprowadzam również kompleksową optymalizację SEO.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
